src/safeposix/filesystem.rs

```diff
@@ -372,6 +389,30 @@ pub fn normpath(origp: interface::RustPathBuf, cage: &Cage) -> interface::RustPa
     newp
 }
 
+pub fn remove_domain_sock(truepath: interface::RustPathBuf) {
+    match metawalkandparent(truepath.as_path()) {
+        //If the file does not exist
+        (None, ..) => { panic!("path does not exist") }
+        //If the file exists but has no parent, it's the root directory
+        (Some(_), None) => { panic!("cannot unlink root directory") }
```

Comment:
Why do we panic rather than throw an exception here? Would this terminate the whole program?

Reply:
This is called during cleanup and is on a list of files we're keeping to cleanup. If the file doesn't exist but we think it does, something went very wrong.

Reply:
By "files" here I mean unix domain sockets, which are files but not like you think of them. We don't persist them from run to run so this function is making sure that they're removed before file persistence.